### PR TITLE
gnome: update to 46.0.

### PR DESCRIPTION
**Pull request by fanyx**
### Missing for `gnome-core`
- [x] `gnome-control-center`
- [x] https://github.com/void-linux/void-packages/pull/49427
- [x] https://github.com/void-linux/void-packages/pull/49446
- [x] https://github.com/void-linux/void-packages/pull/49473
- [x] https://github.com/void-linux/void-packages/pull/49482
- [x] `gjs>=1.80.0`
- [x] https://github.com/void-linux/void-packages/pull/49895

### Missing for `gnome`

- [x] `gnome-remote-desktop-46.2`
  - [x] https://github.com/void-linux/void-packages/pull/49982
  - [x] https://github.com/void-linux/void-packages/pull/49985
  - [x] https://github.com/void-linux/void-packages/pull/50653
- [x] https://github.com/void-linux/void-packages/pull/49940
- [x] https://github.com/void-linux/void-packages/pull/49941
- [x] https://github.com/void-linux/void-packages/pull/49918
- [x] https://github.com/void-linux/void-packages/pull/49945
- [x] https://github.com/void-linux/void-packages/pull/49947
- [x] https://github.com/void-linux/void-packages/pull/49946
- [x] https://github.com/void-linux/void-packages/pull/49957
- [x] https://github.com/void-linux/void-packages/pull/49913
- [x] https://github.com/void-linux/void-packages/pull/49916
- [x] https://github.com/void-linux/void-packages/pull/49639
- [x] https://github.com/void-linux/void-packages/pull/50906
  - [x] https://github.com/void-linux/void-packages/pull/49642
- [x] https://github.com/void-linux/void-packages/pull/49905

@oreo639 @Gottox 

### TODO

- [x] Update `glib` and `glib-networking` dependencies to `2.80.0`

#### Testing the changes
- I tested the changes in this PR:
  - `gnome-core`: **YES**
  - `gnome`: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

**Comment by oreo639** [comment]
> `gnome-control-center` (no 46.0 release yet, [changelog](https://gitlab.gnome.org/GNOME/gnome-control-center/-/raw/main/NEWS))

The 46.0 release changelog is in the gnome-46 branch.
The 46.0 release is also here: https://download.gnome.org/sources/gnome-control-center/46/

**Comment by fanyx** [comment]
https://gitlab.gnome.org/GNOME/gdm/-/commit/2da9b953e3c3931f59d2e312f2deba10ca628146
merged yesterday, not yet in builds though

File: srcpkgs/gnome-core/template [hunk 1]
```diff
@@ -10,27 +10,27 @@ homepage="https://www.gnome.org"
...
+ at-spi2-core>=2.52.0
+ at-spi2-atk>=2.52.0
  cantarell-fonts>=0.301
  dconf>=0.40.0
- evolution-data-server>=3.50.4
- folks>=0.15.6
+ evolution-data-server>=3.52.0
+ folks>=0.15.8
  font-adobe-source-code-pro>=2.030R
  gcr>=3.41.1
```
Comment (fanyx) [inline srcpkgs/gnome-core/template:20]: @oreo639 shouldn't we depend on `gcr4` instead?
all the other packages are switching to `gcr4-devel`

**Comment by oreo639** [comment]
> https://gitlab.gnome.org/GNOME/gdm/-/commit/2da9b953e3c3931f59d2e312f2deba10ca628146
> merged yesterday, not yet in builds though

Just add `.patch` to the end of the url and you can import the patch.

**Comment by fanyx** [comment]
removed `tracker` and `tracker-miners`
tracked in #49473

File: srcpkgs/gdm/template [hunk 2]
```diff
@@ -25,10 +25,10 @@ short_desc="GNOME Display Manager"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Projects/GDM"
-changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/gnome-45/NEWS"
 #changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/main/NEWS"
+changelog="https://gitlab.gnome.org/GNOME/gdm/-/raw/46.0/NEWS/"
```
Comment (oreo639) [inline srcpkgs/gdm/template:29]: This should be `https://gitlab.gnome.org/GNOME/gdm/-/raw/gnome-46/NEWS` (the 46.0 is the tag for the 46.0 release and doesn't contain the changelogs for the whole release series)

And since the gnome-46 branch hasn't been created yet, you can comment it and uncomment the one above.

Comment (fanyx) [reply srcpkgs/gdm/template]: fixed

**Comment by Jaymz665** [comment]
On gnome-core 45.5, a bug was noticed with the display of qt windows and wine windows (something else?).
qt windows do not change size by hovering the cursor over the edge as usual.
Wine too, while the cursor inside the window may stop interacting with the window itself (as if it went beyond the boundaries)
The problem exists on wayland and xorg only at the first login, if you re-register, the problem disappears.
I have now compiled gnome-core 46.0 from your sources and the problem persists. (I am writing through a translator, there may be a misunderstanding)

**Comment by chrysos349** [comment]
@Jaymz665 
Downgrade to `gtk4-4.12.5`, and see if the problem disappears. It did for me. `gtk4-4.14.1` is buggy, imo.

Use the patch below:
```
--- b/srcpkgs/gtk4/template
+++ a/srcpkgs/gtk4/template
@@ -1,6 +1,6 @@
 # Template file for 'gtk4'
 pkgname=gtk4
+version=4.12.5
-version=4.14.1
 revision=1
 build_style=meson
 build_helper="gir"
@@ -30,10 +30,10 @@
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1-or-later"
 homepage="https://www.gtk.org/"
+# changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/main/NEWS"
+changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/gtk-4-12/NEWS"
-changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/main/NEWS"
-# changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/gtk-4-14/NEWS"
 distfiles="${GNOME_SITE}/gtk/${version%.*}/gtk-${version}.tar.xz"
+checksum=28b356d590ee68ef626e2ef9820b2dd21441484a9a042a5a3f0c40e9dfc4f4f8
-checksum=fcefb3f132f8cc4711a9efa5b353c9ae9bb5eeff0246fa74dbc2f2f839b9e308
 
 # Package build options
 build_options="broadway cloudproviders colord cups gir vulkan wayland x11 tracker"
```

P.S. I know `nautilus-46.0` requires `gtk4>=4.13.6`, but works fine with `4.12.5`

**Comment by oreo639** [comment]
@Jaymz665 Can you make a separate issue for this: https://github.com/void-linux/void-packages/issues/new?assignees=&labels=bug%2Cneeds-testing&projects=&template=bug-report.yml

**Comment by Jaymz665** [comment]
@oreo639 yes, of course, but let's first find out exactly what the problem is, because on @chrysos349  advice, I rolled back to gtk4-4.12.5 and the problem disappeared (although the console stopped working). So I think we need to understand where the error lies. My local 45.4 had no errors, the current 45.5 and (honestly stolen templates) on 46.0 have it. So far, it really looks like gtk4, but since you reproduced the error, does it mean there is something else?

File: srcpkgs/gnome-core/template (hunk 1 above)
Comment (oreo639) [reply srcpkgs/gnome-core/template:20]: Not entirely sure why gcr is depended on by gnome-core tbh. Currently we only provide the gcr-ssh-agent executable with gcr3, although that can probably be changed.

imo, leave it for now, although it will probably be removed later.

**Comment by fanyx** [comment]
Updated `glib`, `gvfs` and `tracker` depends

**Comment by fanyx** [comment]
https://github.com/void-linux/void-packages/commit/71c04b6e5670a30710e6dbd96c70237e2f8c4044

guess i'll remove `evolution-data-server` from this, huh?

**Comment by oreo639** [comment]
> guess i'll remove `evolution-data-server` from this, huh?

Yes. You can do `git rebase -i HEAD~19`, then replace `pick` for that commit with `drop`.

**Comment by fanyx** [comment]
Dropped `evolution-data-server` and gcc14.patch for `xdg-desktop-portal-gnome`

**Comment by oreo639** [comment]
Feel free to submit prs for those, most of them can be updated seperately (except for gnome-control-center ofc)

**Comment by fanyx** [comment]
i'll take on `sushi` as it kinda is part of gnome46
`control-center` will be part of this PR, might as well

the rest is probably best left to their own PR and mentioned here

**Comment by oreo639** [comment]
> i'll take on `sushi` as it kinda is part of gnome46

sushi should be a separate PR.

**Comment by fanyx** [comment]
oh yeah, absolutely
but i'll look into it as i want it ready for when we review this `gnome-core` PR as sort of a "gnome 46 release"

the rest can probably be updated separately

**Comment by fanyx** [comment]
Updated `gnome-control-center` and `gnome-online-accounts`

**Comment by fanyx** [comment]
@oreo639 ready for review imo

**Comment by oreo639** [comment]
Not yet, also remove ci skip. Also, gnome-online-accounts requires quite a few rebuilds that are not accounted for.

**Comment by fanyx** [comment]
oh, yeah..
should i add those here?

**Comment by oreo639** [comment]
> should i add those here?

If gnome-online-accounts can be updated separately, probably best to do it separately, but I'll check.

File: srcpkgs/gnome-online-accounts/template [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
 revision=1
 build_style=meson
 build_helper="gir"
 configure_args="$(vopt_bool gir introspection) $(vopt_bool gir vapi)
- -Dmedia_server=true -Dman=true"
+ -Dman=true"
 hostmakedepends="pkg-config gettext glib-devel docbook-xsl libxslt"
-makedepends="gtk+3-devel libwebkit2gtk41-devel json-glib-devel libsecret-devel
- rest-devel gcr-devel mit-krb5-devel libsoup3-devel"
+makedepends="gtk4-devel libwebkit2gtk41-devel json-glib-devel libsecret-devel
```
Comment (oreo639) [inline srcpkgs/gnome-online-accounts/template:10]: libwebkit2gtk41-devel can be removed.

**Comment by fanyx** [comment]
what about all the other revdeps of `gnome-online-accounts`?

**Comment by oreo639** [comment]
> what about all the other revdeps of `gnome-online-accounts`?

`xbps-query -Rs libgoa-backend-1.0.so -p shlib-requires`

**Comment by oreo639** [comment]
> `libgoa-1.0.so.0` is irrelevant?

Should be, the soversion didn't change so it should still be backwards compatible. The main change was that the widgets use gtk4 instead of webkitgtk.

**Comment by fanyx** [comment]
`gtk4-devel` should probably depend on `gtk4-update-icon-cache`

**Comment by chrysos349** [comment]
**gnome-tweaks**

```diff
diff --git a/srcpkgs/gnome-tweaks/template b/srcpkgs/gnome-tweaks/template
index 963940101..9954c5e7f 100644
--- a/srcpkgs/gnome-tweaks/template
+++ b/srcpkgs/gnome-tweaks/template
@@ -1,17 +1,19 @@
 # Template file for 'gnome-tweaks'
 pkgname=gnome-tweaks
-version=45.2
+version=46.0
 revision=1
 build_style=meson
-hostmakedepends="pkg-config gettext"
-makedepends="libhandy1-devel"
-depends="gtk+3 dconf gnome-settings-daemon mutter libnotify python3-gobject libhandy1"
+build_helper=gir
+hostmakedepends="desktop-file-utils gettext gtk4-update-icon-cache pkg-config"
+makedepends="gsettings-desktop-schemas-devel libadwaita-devel
+ python3-gobject-devel"
+depends="dconf gnome-settings-daemon libadwaita mutter python3-gobject"
 short_desc="GNOME tool to customize advanced options"
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="GPL-3.0-or-later, CC0-1.0"
 homepage="https://wiki.gnome.org/Apps/Tweaks"
 distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
-checksum=01bf5e723a1d18b1025401e32b5f64eb29b79e8a17ad010bbbd1e23b137b9069
+checksum=1678786341f664ac4580784222a16409fa9bf5cc7a5f1b46dcefd2aa13ddba31
 replaces="gnome-tweak-tool>=0"
 
 gnome-tweak-tool_package() {
```

**Comment by fanyx** [comment]
> gnome-tweaks

isn't that part of `gnome`, not `gnome-core`?

**Comment by chrysos349** [comment]
i know. gnome-tweaks needs  `gsettings-desktop-schemas>= 46.0`. might as well add it here. let @oreo639 decide.

**Comment by fanyx** [comment]
it builds fine when cherry picking just the `gsettings-desktop-schemas` commit

i created a draft PR (https://github.com/void-linux/void-packages/pull/49925) noting that `gnome-core` should be merged first using your template.
Thanks for providing it <3

File: srcpkgs/gnome/template [hunk 4]
```diff
@@ -25,28 +25,28 @@ depends="
  gnome-music>=${version}
  gnome-remote-desktop>=${version}
  gnome-system-monitor>=${version}
- gnome-text-editor>=${version}
- gnome-tweaks>=42.beta
+ gnome-text-editor>=46.1
```
Comment (oreo639) [inline srcpkgs/gnome/template:28]: This should be `${version}`

File: srcpkgs/gnome/template [hunk 5]
```diff
@@ -25,28 +25,28 @@ depends="
...
+ gnome-tweaks>=${version}
  gnome-weather>=${version}
  gsound>=1.0.3
  gspell>=1.12.2
  iio-sensor-proxy>=3.4
- loupe>=${version}
+ loupe>=46.2
```
Comment (oreo639) [inline srcpkgs/gnome/template:34]: This should be `${version}`

File: srcpkgs/gnome/template [hunk 6]
```diff
@@ -25,28 +25,28 @@ depends="
...
  orca>=${version}
  phodav>=3.0
  power-profiles-daemon>=0.13
- simple-scan>=44.0
- snapshot>=${version}
+ simple-scan>=${version}
+ snapshot>=46.2
```
Comment (oreo639) [inline srcpkgs/gnome/template:39]: Same here.

**Comment by chrysos349** [comment]
**Updates**
 - gnome-maps-46.10
 - gnome-shell-46.1
 - gnome-shell-extensions-46.1
 - gnome-user-docs-46.1
 - mutter-46.1
 - nautilus-46.1

**Comment by fanyx** [comment]
Thanks, will update those.

**Comment by chrysos349** [comment]
**Updates**
 - xdg-desktop-portal-gnome -> 46.1
 - gnome-calendar -> 46.1

**Comment by fanyx** [comment]
gnome-maps, nautilus and gnome-calendar have their own PRs

**Comment by chrysos349** [comment]
> `gnome-remote-desktop-46.1`

also requires:
- libei>= 1.2.0
- polkit>= 122

**Comment by chrysos349** [comment]
**Updates**
 - gnome-boxes-46.1
 - gnome-control-center-46.1

**Comment by chrysos349** [comment]
**Updates**
- gnome-tweaks->46.1
- tracker{,-miners}->3.7.2

**Comment by fanyx** [comment]
Added `tracker{-miners}`
Feel free to open a separate PR, i'd remove the commits

**Comment by chrysos349** [comment]
**Updates**
- evolution-data-server->3.52.2
- gnome-initial-setup->46.2
- gvfs->1.54.1

**Comment by chrysos349** [comment]
**Updates**
 - adwaita-icon-theme-46.2
 - gnome-control-center-46.2
 - gnome-online-accounts-3.50.2
 - gnome-shell-46.2
 - gnome-shell-extensions-46.2
 - mutter-46.2
 - nautilus-46.2
 - xdg-desktop-portal-gnome-46.2

**Comment by fanyx** [comment]
Could you already call this gnome-46.2? :D

**Comment by fanyx** [comment]
@chrysos349 @oreo639 
Do any of you know what's happening with $GNOME_SITE being weird from the GH pipeline?
It seems so weird... Those errors never happen when i build manually

**Comment by fanyx** [comment]
Updates
- gnome-calculator-46.1
- gnome-software-46.2 (optional)
- gnome-text-editor-46.3 (https://github.com/void-linux/void-packages/pull/50587)
- orca-46.1
- rygel-0.42.6

**Comment by chrysos349** [comment]
gmime->3.2.15

File: srcpkgs/gnome-control-center/template [hunk 7]
```diff
@@ -14,17 +14,18 @@ makedepends="ModemManager-devel accountsservice-devel alsa-plugins
  pulseaudio-devel samba-devel upower-devel vala-devel
  libxml2-devel libX11-devel gtk4-devel libadwaita-devel cups-devel libgudev-devel
  libwacom-devel clutter-devel libsecret-devel libcanberra-devel udisks2-devel
- gsound-devel gcr-devel gnutls-devel tecla"
+ gsound-devel gcr4-devel gnutls-devel tecla"
 depends="desktop-file-utils upower colord cups-pk-helper cracklib iso-codes
  gsettings-desktop-schemas hicolor-icon-theme sound-theme-freedesktop
  libgnomekbd ibus-gtk4"
```
Comment (oreo639) [inline srcpkgs/gnome-control-center/template:20]:
```suggestion
 tecla ibus-gtk4"
```

Also, remove libgnomekbd-devel from makedepends.

**Comment by oreo639** [comment]
gmime and at-spi2-core can probably be separate PRs.
Also, no need to comment out the gnome gitlab urls for gmime as that was never the official upstream apparently: https://github.com/jstedfast/gmime/issues/119 (just remove them entirely)

**Comment by oreo639** [comment]
GPaste needs to be updated in this pr.

File: srcpkgs/gnome-control-center/template (hunk 7 above)
Comment (fanyx) [reply srcpkgs/gnome-control-center/template]: Done

**Comment by fanyx** [comment]
> GPaste needs to be updated in this pr.

Why not a separate PR? There is no dependency in gnome on it?

**Comment by oreo639** [comment]
> > GPaste needs to be updated in this pr.
> 
> Why not a separate PR? There is no dependency in gnome on it?

Sure.

**Comment by oreo639** [comment]
Can you add the following patch for gnome-control-center: https://gist.github.com/oreo639/ab7d3cfbf8724af514efeaed326024f7

**Comment by fanyx** [comment]
@oreo639 @chrysos349 
when updating `gnome-remote-desktop` to 46.2 it segfaults instantly.
Any of you know if it is tied in with `mutter` or anything like that? I was running 45 on the rest of my system to test

**Comment by oreo639** [comment]
> when updating `gnome-remote-desktop` to 46.2 it segfaults instantly.

Works for me.

**Comment by fanyx** [comment]
> It almost certainly needs to be updated along with mutter and gdm.

that's what i thought
can we declare the version changes in dependencies or just be prepared to update all the packages at once?

File: srcpkgs/gnome-remote-desktop/template [hunk 8]
```diff
@@ -1,18 +1,20 @@
...
 revision=1
 build_style=meson
+build_helper="gir"
 configure_args="-Drdp=true -Dvnc=true -Dsystemd=false
  -Dsystemd_user_unit_dir=/usr/lib/systemd/user -Dtests=false"
 hostmakedepends="pkg-config gettext glib-devel asciidoc"
 makedepends="glib-devel pipewire-devel libsecret-devel libnotify-devel
- freerdp-devel freerdp-server-devel fuse3-devel libvncserver-devel
- libgudev-devel nv-codec-headers tpm2-tss-devel fdk-aac-devel libei-devel"
+ freerdp3-devel freerdp3-devel fuse3-devel libvncserver-devel
```
Comment (oreo639) [inline srcpkgs/gnome-remote-desktop/template:11]: freerdp3-devel is duplicated here

File: srcpkgs/gnome-remote-desktop/template [hunk 9]
```diff
@@ -1,18 +1,20 @@
 # Template file for 'gnome-remote-desktop'
 pkgname=gnome-remote-desktop
-version=45.1
+version=46.2
 revision=1
 build_style=meson
+build_helper="gir"
```
Comment (oreo639) [inline srcpkgs/gnome-remote-desktop/template:6]: Why did you add this? I don't see GIR/Typelib generation used in this package?

Comment (fanyx) [reply srcpkgs/gnome-remote-desktop/template:6]: oh i saw something about polkit-gobject stuff
but that was resolved by `polkit-devel`, right, i'll remove it again

File: srcpkgs/gnome-remote-desktop/template (hunk 8 above)
Comment (fanyx) [reply srcpkgs/gnome-remote-desktop/template:11]: oops...

**Comment by oreo639** [comment]
Replace `webp-pixbuf-loader` in gnome-backgrounds with `libjxl-plugins`

File: srcpkgs/xdg-desktop-portal-gnome/template [hunk 10]
```diff
@@ -14,7 +14,7 @@ short_desc="GNOME portal backend for xdg-desktop-portal"
 maintainer="oreo639 <oreo6391@gmail.com>"
 license="LGPL-2.1-or-later"
 homepage="https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome"
-changelog="https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/raw/gnome-45/NEWS"
 #changelog="https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/raw/main/NEWS"
+changelog="https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/raw/46.0/NEWS"
```
Comment (oreo639) [inline srcpkgs/xdg-desktop-portal-gnome/template:18]: This should be `https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/raw/gnome-46/NEWS`

File: srcpkgs/gnome-tweaks/template [hunk 11]
```diff
@@ -1,17 +1,19 @@
 # Template file for 'gnome-tweaks'
 pkgname=gnome-tweaks
-version=45.2
+version=46.1
 revision=1
 build_style=meson
-hostmakedepends="pkg-config gettext"
-makedepends="libhandy1-devel"
-depends="gtk+3 dconf gnome-settings-daemon mutter libnotify python3-gobject libhandy1"
+build_helper=gir
```
Comment (oreo639) [inline srcpkgs/gnome-tweaks/template:6]: Not sure this is necessary. (I don't see any GIR/Typelib generation)
Let me know if it is.

**Comment by fanyx** [comment]
thx for the review
fixed those

File: srcpkgs/mutter/template [hunk 12]
```diff
@@ -18,12 +18,11 @@ short_desc="Wayland display server, X11 window manager and compositor library"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Projects/Mutter/"
-#changelog="https://gitlab.gnome.org/GNOME/mutter/-/raw/gnome-45/NEWS"
```
Comment (oreo639) [inline srcpkgs/mutter/template:21]: Add this back, uncomment it and make it `https://gitlab.gnome.org/GNOME/mutter/-/raw/gnome-46/NEWS`

**Comment by fanyx** [comment]
how the...
when i push gnome commits, the builds always fail with fetching distfile issues lol

**Comment by oreo639** [comment]
> when i push gnome commits, the builds always fail with fetching distfile issues lol

It didn't always happen, it happened for a period probably due to rate limiting on GH Actions pulling distfiles from gnome. Whether it was triggered by us or someone else idk.

**Comment by fanyx** [comment]
yeah ik its not my fault
i was just annoyed by it ^^

**Comment by oreo639** [comment]
@fanyx Is there anything left that needs to be done, otherwise I can merge this?

**Comment by fanyx** [comment]
everything of note has been done, but i'm in no way qualified enough to be sure i've caught everything

i believe we have been running this for a while without issues, so
